### PR TITLE
json: Build Gstreamer-vaapi as a module

### PR DESCRIPTION
**Pull request by EliasOfWaffle**
Gstreamer is now in release 1.19 i think that is possible to be stable for use now, but is needed more testing

### CODE REVIEW

File: org.gnome.Epiphany.json [hunk 1]
```diff
@@ -25,6 +25,18 @@
...
+        {
+            "name": "gstreamer-vaapi",
+            "buildsystem": "meson",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/GStreamer/gstreamer-vaapi",
+                    "commit": "c3ddb29cb2860374f9efbed495af7b0eead08312"
+                }
+            }
```
Comment (A6GibKm) [inline org.gnome.Epiphany.json:38]: Extra }

File: org.gnome.Epiphany.json [hunk 2]
```diff
@@ -25,6 +25,18 @@
...
                 }
             ],
+    "modules": [
+        {
+            "name": "gstreamer-vaapi",
+            "buildsystem": "meson",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/GStreamer/gstreamer-vaapi",
```
Comment (A6GibKm) [inline org.gnome.Epiphany.json:35]: Please use the gitlab instance instead of a fork

Comment (A6GibKm) [inline org.gnome.Epiphany.json:35]: Additionally, please use the mono repo https://gitlab.freedesktop.org/gstreamer/gstreamer/ and pick the gstreamer-vaapi submodule from there, see how Decoder for example does it

File: org.gnome.Epiphany.json [hunk 3]
```diff
@@ -25,6 +25,18 @@
...
+                    "commit": "c3ddb29cb2860374f9efbed495af7b0eead08312"
```
Comment (A6GibKm) [inline org.gnome.Epiphany.json:36]: Please use a tag instead, its easier to know what version is being used like that.

File: org.gnome.Epiphany.json (hunk 2 above)
Comment (EliasOfWaffle) [reply org.gnome.Epiphany.json]: oh thanks i will make it now

**Comment by mcatanzaro** [comment]
This won't do anything: WebKit denylists gstreamer-vaapi to prevent it from being used, because it's too broken. So I'm going to close this. Um, although I notice you just closed it yourself, so maybe you noticed? :)

If you want hardware acceleration you need to forget about gstreamer-vaapi and focus on the va element in gst-plugins-bad instead. In GStreamer 1.22 it is downranked, but in GStreamer 1.24 it should be enabled by default.

Then the next problem is that the web process won't have permission to access decoders. That would require fixing https://github.com/flatpak/flatpak/issues/5524 in flatpak to add a new permission for video decoder access. Once that's fixed, we can change WebKit to pass the new permission to `flatpak-spawn` and then change the Epiphany manifest to also use the new permission. So the focus now needs to be on https://github.com/flatpak/flatpak/issues/5524.

**Comment by mcatanzaro** [comment]
See https://bugs.webkit.org/show_bug.cgi?id=240664 for more info.

**Pull request closed without merging**